rfcs/45-stageless.md

```diff
@@ -0,0 +1,727 @@
+By default, systems are added to the main schedule.
+You can control this by adding the `.in_set_schedule(ScheduleLabel::Variant)` system descriptor to your system or system set.
+
+#### Startup systems
+
+Startup systems are stored in their own schedule, with the `CoreSchedule::Startup` label.
+When using the runner added by `MinimalPlugins` and `DefaultPlugins`, this schedule will run exactly once on app startup.
+
+You can add startup systems with the `.add_startup_system(on_startup)` method on `App`, which is simply sugar for `.add_system(on_startup.in_set_schedule(CoreSchedule::Startup))`.
+
+### System configuration
+
+Each system has a few configurable parameters:
+
+- it may have **ordering constraints**, causing it to run before or after other systems
+  - there are several subtly different types of ordering constraints, see the next section for details
+  - e.g. `.add_system(player_controls.before(GameSets::Physics).after(CoreSets::Input))`
+- it may have one or more **run criteria** attached
+  - a system is only executed if all of its run criteria return `true`
+  - **states** are a special, more complex pattern that use run criteria to determine whether or not a system should run in a current state
+  - e.g. `.add_system(physics.run_if(resource_equals(EnablePhysics(true))))`
+- it may have one or more **labels**, allowing other systems to refer to it and enabling mass-configuration
+  - e.g. `.add_system(physics.in_set(GameSets::Physics)`
+
+This configuration is additive: adding another ordering constraint, run criteria or set does not replace the existing configuration.
+
+System configuration can be stored in a `SystemConfig` struct.
+This can be useful to reuse, compose and quickly apply complex configuration strategies, before applying these strategies to sets or individual systems.
+
+### System set configuration
+
+Each system set has its own associated `SystemConfig`, stored in the corresponding `Schedule`. When a system set is configured, its configuration is effectively passed down to every system under it. For example, ordering the set `A` to run before the set `B` specifies that all systems in `A` will run before any system in `B` does.
+
+You can add many systems to the same set at once using the `App::add_systems(systems: impl SystemIterator)` method.
+
+```rust
+#[derive(SystemLabel)]
+enum Physics {
+    Forces,
+    CollisionDetection,
+    CollisionHandling,
+}
+
+impl Plugin for PhysicsPlugin{
+    fn build(app: &mut App){
+        // Within each frame, physics logic needs to occur after input handling, but before rendering
+        let mut common_physics_config = SystemConfig::new().after(InputSet::ReadInputHandling).before(CoreSet::Rendering);
+
+        app
+        // We can reuse this shared configuration on each of our sets
+        .configure_set(Physics::Forces.configure(common_physics_config).before(Physics::CollisionDetection))

```

Comment:
What type is accepted by `configure_set`? If it is SystemConfig, how do we choose which set is being configured here:

```rust
app.configure_set(Physics::Forces.configure(common_physics_config).in_set(SomeOtherSet))
```

Should adding a set to another set be allowed? Is there some sort of `SystemConfig::configured_set: Option<Box<dyn SystemLabel>>`? Or does configure_set take a `SetConfig` type?